apiserver/facades/agent/uniter/uniter.go

```diff
@@ -840,7 +840,7 @@ func (u *UniterAPI) waitForCacheUnit(tag names.UnitTag, condition func(cu cache.
 			return errors.New("timed out waiting for change to be reflected in cache")
```

Comment:
I wonder if we should actually be returning NotFound here.

Because if you consider the case where the unit really is gone now, shouldn't that be the interaction?